tls-client/main.cpp

```diff
@@ -132,6 +133,8 @@ class HelloHTTPS {
         mbedtls_x509_crt_free(&_cacert);
         mbedtls_ssl_free(&_ssl);
         mbedtls_ssl_config_free(&_ssl_conf);
+        _tcpsocket->close();
```

Comment:
```close()``` is a harmless method and if you see its implementation, it checks if the socket is open or not. Moving it to the destructor add a lot of simplicity as we don't have to call it every time we ```return``` from ```startTest``` or use ```goto``` statements.